download_action.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+)
+
+type DownloadAction struct {
+	BaseAction `yaml:",inline"`
+	Url        string // URL for downloading
+	Filename   string // File name, overrides the name from URL. TODO: optional
```

Comment:
I'd leave this one out in the initial commits. Whether it's useful will probably depend on how we end up using things and it'll clash with unpacking?